process/block/metablock.go

```diff
@@ -1668,6 +1653,42 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 	return body, nil
 }
 
+// ApplyValidatorStatistics calculates the new statistics root hash and updates the metachain header with the new value
+func (mp *metaProcessor) ApplyValidatorStatistics(header data.HeaderHandler) error {
```

Comment:
this way everybody is creating outside of consensus the validator statistics roothash and ads to the header. The salivadot statistics changes could be done for the last header, not the current one. That way to code remains as it is.

Reply:
The only change needed is to call validatorstatistics.UpdatePeerState with the current header, what we have in blockChain.GetCurrentHeader.